FreeRTOS/FreeRTOS-Cellular-Interface
# Parsing CREG/CGREG/CEREG separately.

Parsing CREG/CGREG/CEREG separately.

Description
-----------

Like [FreeRTOS forum post](https://forums.freertos.org/t/cellular-interface-cgreg-parsing/21837) mentioned, the cellular interface doesn't handle CGREG RAC correctly. To parse all these reg status better, this PR separates the parsing flow into different functions.
- Separate register status parsing for CREG/CGREG/CEREG.
- Early break if any parsing error happens.
- Adjust UT cases with the change.

Test Steps
-----------

1. UT
    1. Create UT new test cases
    1. Execute UT
    1. Achieve 100% coverage
1. Real target test from forum customer.
    - https://forums.freertos.org/t/cellular-interface-cgreg-parsing/21837/4

Checklist:
----------



Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
